eth/tracers/live/supply.go

```diff
@@ -0,0 +1,245 @@
+
+	s.delta.Number = b.NumberU64()
+	s.delta.Hash = b.Hash()
+	s.delta.ParentHash = b.ParentHash()
+
+	// Initialize supply with total allocation in genesis block
+	for _, account := range alloc {
+		s.delta.Delta.Add(s.delta.Delta, account.Balance)
+	}
+
+	out, _ := json.Marshal(s.delta)
+	s.logger.Println(string(out))
+}
+
+func (s *supply) OnBalanceChange(a common.Address, prevBalance, newBalance *big.Int, reason tracing.BalanceChangeReason) {
+	diff := new(big.Int).Sub(newBalance, prevBalance)
+
+	// NOTE: don't handle "BalanceIncreaseGenesisBalance" because it is handled in OnGenesisBlock
+	switch reason {
+	case tracing.BalanceIncreaseRewardMineUncle:
+	case tracing.BalanceIncreaseRewardMineBlock:
+		s.delta.Reward.Add(s.delta.Reward, diff)
+	case tracing.BalanceIncreaseWithdrawal:
+		s.delta.Withdrawals.Add(s.delta.Withdrawals, diff)
+	case tracing.BalanceDecreaseSelfdestructBurn:
+		// BalanceDecreaseSelfdestructBurn is non-reversible as it happens
+		// at the end of the transaction.
+		s.delta.Burn.Sub(s.delta.Burn, diff)
+	default:
+		return
+	}
+
+	s.delta.Delta.Add(s.delta.Delta, diff)
+}
+
+func (s *supply) OnTxStart(vm *tracing.VMContext, tx *types.Transaction, from common.Address) {
+	s.txCallstack = make([]supplyTxCallstack, 0, 1)
+}
+
+// internalTxsHandler handles internal transactions burned amount
+func (s *supply) internalTxsHandler(call *supplyTxCallstack) {
+	// Handle Burned amount
+	if call.burn != nil {
+		s.delta.burn(call.burn)
+	}
+
+	if len(call.calls) > 0 {
+		// Recursivelly handle internal calls
+		for _, call := range call.calls {
+			callCopy := call
+			s.internalTxsHandler(&callCopy)
+		}
+	}
+}
+
+func (s *supply) OnEnter(depth int, typ byte, from common.Address, to common.Address, input []byte, gas uint64, value *big.Int) {
+	call := supplyTxCallstack{
+		calls: make([]supplyTxCallstack, 0),
+	}
+
+	// This is a special case of burned amount which has to be handled here
+	// which happens when type == selfdestruct and from == to.
+	if vm.OpCode(typ) == vm.SELFDESTRUCT && from == to && value.Cmp(common.Big0) == 1 {
+		call.burn = value
+	}
+
+	// Append call to the callstack, so we can fill the details in CaptureExit
+	s.txCallstack = append(s.txCallstack, call)
+}
+
+func (s *supply) OnExit(depth int, output []byte, gasUsed uint64, err error, reverted bool) {
+	if depth == 0 {
+		// No need to handle Burned amount if transaction is reverted
+		if !reverted {
+			s.internalTxsHandler(&s.txCallstack[0])
+		}
+		return
+	}
+
+	size := len(s.txCallstack)
+	if size <= 1 {
+		return
+	}
+	// Pop call
+	call := s.txCallstack[size-1]
+	s.txCallstack = s.txCallstack[:size-1]
+	size -= 1
+
+	// In case of a revert, we can drop the call and all its subcalls.
+	// Caution, that this has to happen after popping the call from the stack.
+	if reverted {
+		return
+	}
+	s.txCallstack[size-1].calls = append(s.txCallstack[size-1].calls, call)
+}
+
+func (s *supply) OnTerminate() {
+	s.loggerOutputFile.Close()
```

Comment:
Let's check the error and log a warning

Reply:
Solved by [f65a19a](https://github.com/ethereum/go-ethereum/pull/29347/commits/f65a19af15fd43bda8b80484735001dd332791aa)